README.md
# Contact information block

## About

This module will allow you to display your e-store's contact information in a customizable block.

## Contributing

PrestaShop modules are open-source extensions to the PrestaShop e-commerce solution. Everyone is welcome and even encouraged to contribute with their own improvements.

### Requirements

Contributors **must** follow the following rules:

* **Make your Pull Request on the "dev" branch**, NOT the "master" branch.
* Do not update the module's version number.
* Follow [the coding standards][1].

### Process in details

Contributors wishing to edit a module's files should follow the following process:

1. Create your GitHub account, if you do not have one already.
2. Fork the blockcontactinfos project to your GitHub account.
3. Clone your fork to your local machine in the ```/modules``` directory of your PrestaShop installation.
4. Create a branch in your local clone of the module for your changes.
5. Change the files in your branch. Be sure to follow [the coding standards][1]!
6. Push your changed branch to your fork in your GitHub account.
7. Create a pull request for your changes **on the _'dev'_ branch** of the module's project. Be sure to follow [the commit message norm][2] in your pull request. If you need help to make a pull request, read the [Github help page about creating pull requests][3].
8. Wait for one of the core developers either to include your change in the codebase, or to comment on possible improvements you should make to your code.

That's it: you have contributed to this open-source project! Congratulations!

[1]: http://doc.prestashop.com/display/PS16/Coding+Standards
[2]: http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message
[3]: https://help.github.com/articles/using-pull-requests


CHANGELOG.txt
2016-08-12 14:55:00 +0200	// Add website field
2014-04-22 18:57:06 +0200	// Changelog updated
2014-03-24 11:43:28 +0100	/ MO blockcontactinfos : ps_versions_compliancy modified (1.5.6.1 => 1.6)
2014-03-24 11:01:54 +0100	// MO blockcontactinfos/ : ps_versions_compliancy added
2014-03-20 14:22:32 +0100	Initial commit


index.php
<?php
/*
* 2007-2016 PrestaShop
*
* NOTICE OF LICENSE
*
* This source file is subject to the Academic Free License (AFL 3.0)
* that is bundled with this package in the file LICENSE.txt.
* It is also available through the world-wide-web at this URL:
* http://opensource.org/licenses/afl-3.0.php
* If you did not receive a copy of the license and are unable to
* obtain it through the world-wide-web, please send an email
* to license@prestashop.com so we can send you a copy immediately.
*
* DISCLAIMER
*
* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
* versions in the future. If you wish to customize PrestaShop for your
* needs please refer to http://www.prestashop.com for more information.
*
*  @author PrestaShop SA <contact@prestashop.com>
*  @copyright  2007-2016 PrestaShop SA
*  @license    http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
*  International Registered Trademark & Property of PrestaShop SA
*/

header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
header('Last-Modified: '.gmdate('D, d M Y H:i:s').' GMT');

header('Cache-Control: no-store, no-cache, must-revalidate');
header('Cache-Control: post-check=0, pre-check=0', false);
header('Pragma: no-cache');

header('Location: ../../');
exit;


blockcontactinfos.php
<?php
/*
* 2007-2016 PrestaShop
*
* NOTICE OF LICENSE
*
* This source file is subject to the Academic Free License (AFL 3.0)
* that is bundled with this package in the file LICENSE.txt.
* It is also available through the world-wide-web at this URL:
* http://opensource.org/licenses/afl-3.0.php
* If you did not receive a copy of the license and are unable to
* obtain it through the world-wide-web, please send an email
* to license@prestashop.com so we can send you a copy immediately.
*
* DISCLAIMER
*
* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
* versions in the future. If you wish to customize PrestaShop for your
* needs please refer to http://www.prestashop.com for more information.
*
*  @author PrestaShop SA <contact@prestashop.com>
*  @copyright  2007-2016 PrestaShop SA

*  @license    http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
*  International Registered Trademark & Property of PrestaShop SA
*
*/

if (!defined('_CAN_LOAD_FILES_'))
	exit;

class Blockcontactinfos extends Module
{
	protected static $contact_fields = array(
		'BLOCKCONTACTINFOS_COMPANY',
		'BLOCKCONTACTINFOS_ADDRESS',
		'BLOCKCONTACTINFOS_PHONE',
		'BLOCKCONTACTINFOS_EMAIL',
		'BLOCKCONTACTINFOS_WEBSITE',
	);

	public function __construct()
	{
		$this->name = 'blockcontactinfos';
		$this->author = 'PrestaShop';
		$this->tab = 'front_office_features';
		$this->version = '1.2.1';

		$this->bootstrap = true;
		parent::__construct();

		$this->displayName = $this->l('Contact information block');
		$this->description = $this->l('This module will allow you to display your e-store\'s contact information in a customizable block.');
		$this->ps_versions_compliancy = array('min' => '1.6', 'max' => '1.6.99.99');
	}

	public function install()
	{
		Configuration::updateValue('BLOCKCONTACTINFOS_COMPANY', Configuration::get('PS_SHOP_NAME'));
		Configuration::updateValue('BLOCKCONTACTINFOS_ADDRESS', trim(preg_replace('/ +/', ' ', Configuration::get('PS_SHOP_ADDR1').' '.Configuration::get('PS_SHOP_ADDR2')."\n".Configuration::get('PS_SHOP_CODE').' '.Configuration::get('PS_SHOP_CITY')."\n".Country::getNameById(Configuration::get('PS_LANG_DEFAULT'), Configuration::get('PS_SHOP_COUNTRY_ID')))));
		Configuration::updateValue('BLOCKCONTACTINFOS_PHONE', Configuration::get('PS_SHOP_PHONE'));
		Configuration::updateValue('BLOCKCONTACTINFOS_EMAIL', Configuration::get('PS_SHOP_EMAIL'));
		Configuration::updateValue('BLOCKCONTACTINFOS_WEBSITE', Configuration::get('PS_SHOP_WEBSITE'));
		$this->_clearCache('blockcontactinfos.tpl');
		return (parent::install() && $this->registerHook('header') && $this->registerHook('footer'));
	}

	public function uninstall()
	{
		foreach (Blockcontactinfos::$contact_fields as $field)
			Configuration::deleteByName($field);
		return (parent::uninstall());
	}

	public function getContent()
	{
		$html = '';
		if (Tools::isSubmit('submitModule'))
		{
			foreach (Blockcontactinfos::$contact_fields as $field)
				Configuration::updateValue($field, Tools::getValue($field));
			$this->_clearCache('blockcontactinfos.tpl');
			$html = $this->displayConfirmation($this->l('Configuration updated'));
		}

		return $html.$this->renderForm();
	}

	public function hookHeader()
	{
		$this->context->controller->addCSS(($this->_path).'blockcontactinfos.css', 'all');
	}

	public function hookFooter($params)
	{
		if (!$this->isCached('blockcontactinfos.tpl', $this->getCacheId()))
			foreach (Blockcontactinfos::$contact_fields as $field)
				$this->smarty->assign(strtolower($field), Configuration::get($field));
		return $this->display(__FILE__, 'blockcontactinfos.tpl', $this->getCacheId());
	}

	public function renderForm()
	{
		$fields_form = array(
			'form' => array(
				'legend' => array(
					'title' => $this->l('Settings'),
					'icon' => 'icon-cogs'
				),
				'input' => array(
					array(
						'type' => 'text',
						'label' => $this->l('Company name'),
						'name' => 'BLOCKCONTACTINFOS_COMPANY',
					),
					array(
						'type' => 'textarea',
						'label' => $this->l('Address'),
						'name' => 'BLOCKCONTACTINFOS_ADDRESS',
					),
					array(
						'type' => 'text',
						'label' => $this->l('Phone number'),
						'name' => 'BLOCKCONTACTINFOS_PHONE',
					),
					array(
						'type' => 'text',
						'label' => $this->l('Email'),
						'name' => 'BLOCKCONTACTINFOS_EMAIL',
					),
					array(
						'type' => 'text',
						'label' => $this->l('Website'),
						'name' => 'BLOCKCONTACTINFOS_WEBSITE',
					),
				),
				'submit' => array(
					'title' => $this->l('Save')
				)
			),
		);

		$helper = new HelperForm();
		$helper->show_toolbar = false;
		$helper->table =  $this->table;
		$lang = new Language((int)Configuration::get('PS_LANG_DEFAULT'));
		$helper->default_form_language = $lang->id;
		$helper->allow_employee_form_lang = Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') ? Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') : 0;
		$this->fields_form = array();

		$helper->identifier = $this->identifier;
		$helper->submit_action = 'submitModule';
		$helper->currentIndex = $this->context->link->getAdminLink('AdminModules', false).'&configure='.$this->name.'&tab_module='.$this->tab.'&module_name='.$this->name;
		$helper->token = Tools::getAdminTokenLite('AdminModules');
		$helper->tpl_vars = array(
			'fields_value' => array(),
			'languages' => $this->context->controller->getLanguages(),
			'id_language' => $this->context->language->id
		);
		foreach (Blockcontactinfos::$contact_fields as $field)
			$helper->tpl_vars['fields_value'][$field] = Tools::getValue($field, Configuration::get($field));
		return $helper->generateForm(array($fields_form));
	}
}
